socorro/external/postgresql/products.py

```diff
@@ -14,11 +18,131 @@
 logger = logging.getLogger("webapi")
 
 
-class Products(PostgreSQLBase):
+class SmartDate(object):
+
+    def clean(self, value):
+        if any(itertools.imap(value.startswith, ('>=', '<='))):
+            operator_ = value[:2]
+            value = value[2:]
+        elif any(itertools.imap(value.startswith, ('=', '>', '<'))):
+            operator_ = value[:1]
+            value = value[1:]
+        else:
+            operator_ = '='
+        return (operator_, string_to_datetime(value).date())
+
```

Comment:
@adngdb So this is the new cool thing. I changed `external_common.parse_arguments` so that one of the "types" you can pass in can instead be a class instance that has a `clean` method. 
Its job is to return the value (doesn't have to be a tuple) or raise a `ValueError` (or `BadArgumentError`).